### PR TITLE
feat(early-kickout): tune production kickout thresholds

### PR DESCRIPTION
**Pull request by stedfn**
Tunes the early-kickout thresholds to the values settled in the protocol-status meeting: a 1000-block grace window at the start of each epoch, at least 100 missed chunks, and under 80% chunk production. Behind `ProtocolFeature::EarlyKickout` (v152, nightly only); off the feature the gate is false and behavior is unchanged.

The miss floor rises from 20 to 100. The per-producer sample-size floor (`EARLY_KICKOUT_MINIMUM_OBSERVED_BLOCKS = 50`) is removed and replaced by a start-of-epoch grace: no producer is blacklisted until the anchor is at least 1000 blocks into its epoch, so upgrade-restart downtime early in an epoch does not trigger kickout. The grace is a height grace (anchor height minus epoch-start height), enforced in the shared `blacklist_for_epoch` gate that both the record-block seeder and the `get_chunk_producer_blacklist` accessor call, so the two cannot drift.

Both callers resolve the epoch start defensively: at genesis the `EpochStart` row does not exist yet, so a missing start is treated as "just started" (grace, empty blacklist); any other lookup error propagates rather than being masked as an empty blacklist.

Tests recalibrate the blacklist math for the 100-miss floor, grow the integration drives past the 1000-block grace, add a grace-boundary test that also checks the seeder and accessor agree at the first active anchor, and make the epoch-boundary-reset test non-vacuous under the grace.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
...
 * Fix a bug in `receiver` verification for a `DeterministicStateInitAction` inside a `DelegateAction` that made it impossible to create deterministic accounts through meta transactions. ([#15812](https://github.com/near/nearcore/pull/15812))
 * Stabilized gas keys ([NEP-611](https://github.com/near/NEPs/pull/611)): a new access key type that carries a pre-funded NEAR balance and up to 1,024 independent nonce sequences. Transactions signed with a gas key pay gas costs from the key's balance rather than the signer's account balance. ([#15183](https://github.com/near/nearcore/pull/15183))
+* nightly mid-epoch chunk-producer kickout (gated at protocol v152): a chunk producer that stops producing is excluded from chunk-producer sampling within the same epoch, reassigning its slots to other producers instead of waiting for the epoch boundary. A producer is kicked once it is past a 1000-block start-of-epoch grace window and has both missed at least 100 chunks and produced under 80% of its expected chunks. ([#15843](https://github.com/near/nearcore/pull/15843))
```
Comment (Trisfald) [inline CHANGELOG.md]: Can we merge this with the existing entry for early kickout? ( we had one right?)

File: chain/epoch-manager/src/tests/early_kickout.rs [hunk 2]
```diff
@@ -394,34 +408,119 @@ fn early_kickout_attribution_does_not_flap() {
...
+    // epoch 1, so the epoch mismatch resets the blacklist to empty.
+    let bl_boundary = handle.get_chunk_producer_blacklist(&h[boundary_idx]).unwrap();
+    assert!(bl_boundary.is_empty(), "epoch boundary must reset blacklist, got {bl_boundary:?}");
+}
+
+// Start-of-epoch grace: with the down node already miss-heavy, the accessor stays empty until
+// the anchor is at least EARLY_KICKOUT_EPOCH_GRACE_BLOCKS into the epoch, then blacklists it.
+#[cfg(feature = "nightly")]
+#[test]
+fn get_chunk_producer_blacklist_respects_epoch_grace() {
```
Comment (Trisfald) [inline chain/epoch-manager/src/tests/early_kickout.rs:447]: Internal comments are on the heavy side for `get_chunk_producer_blacklist_respects_epoch_grace` and `get_chunk_producer_blacklist_resets_on_epoch_boundary`. I'd prefer seeing a single, well stated, general comment on top of the function explaining the test

File: chain/epoch-manager/src/tests/early_kickout.rs [hunk 3]
```diff
@@ -233,6 +236,12 @@ fn record_block_with_mask(
 /// exactly on the heights where `target` is the scheduled producer. The result:
 /// `target` accumulates 0 produced / many expected (blacklist candidate) while the
 /// other producer stays at 100%. Returns the recorded block hashes (index = height).
+///
+/// Stable-only: it uses the *plain* height sampler, so once the grace window lifts and the
```
Comment (Trisfald) [inline chain/epoch-manager/src/tests/early_kickout.rs]: I don't get why we mention "stable-only " behavior here, if the function is gated for nightly

File: CHANGELOG.md (hunk 1 above)
Comment (stedfn) [reply CHANGELOG.md]: I decided to remove them until the feature gets stabilized / do it in a separate PR at the end because they keep changing

Comment (Trisfald) [reply CHANGELOG.md]: Okay 👌

File: chain/epoch-manager/src/tests/early_kickout.rs (hunk 3 above)
Comment (stedfn) [reply chain/epoch-manager/src/tests/early_kickout.rs]: it is gated for `not` nightly `#[cfg(not(feature = "nightly"))]`, or do you mean something else?

Comment (Trisfald) [reply chain/epoch-manager/src/tests/early_kickout.rs]: Right 😅 

Then maybe we don't need the `On nightly` part. Anyway, its a nit, the feature will be stabilized soon

**Review by Trisfald — APPROVED** [review]

**Review by jancionear — APPROVED** [review]
